### DISCUSSION TITLE
Icon's Shape Poll

### DISCUSSION THREAD

**Discussion by daiyam** [VSCodium/vscodium#1429] [Polls]
Vote for VSCodium's next icon!

Here, you can vote for the icon's shape.

**Don't forget to vote for the color at https://github.com/VSCodium/vscodium/discussions/1430.**

The vote ends the 1st April at midnight.

The final icon will the combination of the winners.

|        |                                                           codium                                                            |                                                              c                                                              |                                                           derive                                                            |                                                         simplified                                                          |                                                            style                                                            |
| ------ |:---------------------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------------------------------------------------:|
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| blue   | <img src="https://user-images.githubusercontent.com/587742/222109292-66ca04c1-b5f2-467a-9b23-eed23bda5027.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109252-c693837c-3743-4217-8ffe-7240a08b6455.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109310-99b1bd35-b7d5-4030-8335-66c523ab3b0f.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109325-0ac03458-1e17-4b5d-83a4-fe6e637fd89c.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109343-ac773ce3-cdf7-43a1-89d9-68090ecccef5.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| green  | <img src="https://user-images.githubusercontent.com/587742/222109297-7fc9682e-e1ae-4509-953c-2df39337fe10.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109273-1234290e-ba94-41bd-be1b-4591e129d3eb.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109316-2c381c96-20e6-407b-a133-d74e2ed1523b.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109330-cd527516-8223-4da1-a3ed-fb87aba39c39.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109348-0acd40b6-2193-44db-8029-bc8f008b3c50.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| yellow | <img src="https://user-images.githubusercontent.com/587742/222109306-a657bea5-b7a8-4893-8513-b923024d7a43.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109289-7dc7f9f7-60f2-4315-9867-758d9cc89ebb.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109322-c894be1e-7484-443c-b308-d2a73d7e19e3.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109339-6a9cee57-5f54-411e-9228-383a67ba3d7a.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109359-7f1d5547-9299-4c22-a7cc-0934878d00f5.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| orange | <img src="https://user-images.githubusercontent.com/587742/222109298-f0c8b58a-e383-41d3-a821-b05790ba4d68.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109275-f7b1321e-bcb6-4de8-b5f6-2b2e144c2908.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109317-a0c5da53-224d-4813-85d2-6c99f2ad86e7.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109333-6df8a2b7-1d33-4ed7-b264-82c980e857f3.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109350-a381ccd8-0f87-4317-95cf-436afed4d6a9.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| red    | <img src="https://user-images.githubusercontent.com/587742/222109303-e263e5c4-79bc-41e8-a60e-10077022fef1.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109285-13541083-b9a2-4802-9593-53fa76a41093.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109320-d8a892aa-4f7c-4511-93e4-72d32df9b539.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109337-845011c1-915b-4ee1-ba4d-c5d6a2c98cec.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109356-4d99be8b-e7f2-4bfc-bcd7-9d426b010a86.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| purple | <img src="https://user-images.githubusercontent.com/587742/222109299-1334dc4e-0a11-4561-88c1-6b03edd33be3.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109280-6c25e655-5661-47ea-baaf-8c092a8ea667.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109319-4888512e-a81d-4c25-a652-1bc507e81aee.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109334-47d43ee7-76c6-468e-91f6-cf3a2b8640ac.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109352-2e771538-a0f9-4059-aedd-53eedc577987.png" width="128"> |
|        |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |                                                                                                                             |
| brown  | <img src="https://user-images.githubusercontent.com/587742/222109294-828fa37b-e189-4cfb-a4a6-f64ee66bb251.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109270-10b682d2-6010-45c3-ae31-e04f53d141bc.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109313-2bc220e6-91d2-4201-bd59-b7283ef227ae.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109327-e305d37c-268b-4a5f-a0de-98cb77a0d334.png" width="128"> | <img src="https://user-images.githubusercontent.com/587742/222109346-3d371d70-46d1-4b00-bf56-f77ff2463652.png" width="128"> |

**Don't forget to vote for the color at https://github.com/VSCodium/vscodium/discussions/1430.**

**Comment by GitMensch** [comment]
I'd vote for simplified with not that thick lines...

**Reply by pmahend1** [reply]
Without that much thickness it would be hardly visible in Taskbars,

**Reply by GitMensch** [reply]
I see that this isn't easy... the issue here: when used in small sizes codium (=old) is white with some blue distortion, while simplified is blue splash, because the distance of the blue lines is not big enough; making it one or two sizes thinner likely would solve that, and also look a bit more "nice" and more similar to the old one.

**Reply by Hate9** [reply]
@GitMensch ngl I think you're just straight-up wrong. Simplified looks entirely fine to me even at 16x16.

To actually back that up, here's Simplified at 16x16:
![vscodium-simplified-16x16](https://user-images.githubusercontent.com/4684153/222927839-785c608b-91b8-4303-ab5a-25e452042003.png)

And here's Codium at 16x16:
![vscodium-normal-16x16](https://user-images.githubusercontent.com/4684153/222927840-0d7b31a9-61e2-4882-b144-084d2c372a9d.png)

Like, you can just look at that, and one of those is *clearly* better than the other. If you think simplified looks like a blue splash, you're either looking at it at *ridiculously* small sizes, *or* you have vision problems, and in either case, that's not a problem with the logo.

**Reply by GitMensch** [reply]
:-) Try again with a plain small circle instead of the square and it gets worse...
I'd say both are not ideal but that can also be my personal issue; I'll live with whatever the project will use.

**Reply by FloraCanou** [reply]
The detail can use some improvement, I think. There's some nuanced thickness variation in the original codium and they need to be present in the simplified design too.

**Comment by AuraHerreroRuiz** [comment]
I'd like the voting to be a process of elimination. I like "codium" the most, but I like the simplified too, more than the other ones.

**Reply by GitMensch** [reply]
I see the point but fear that this takes too much time to get final results...

**Reply by AuraHerreroRuiz** [reply]
That is true. The ideal would be being able to vote for the ones you like, but I assume GitHub only lets you create polls that allow for one choice.

**Reply by daiyam** [reply]
Ya, the polls are really limited.

**Comment by evur** [comment]
I think a circle would be better here. (see https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-4342208 and https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-4362865)

**Reply by evur** [reply]
Sorry for the ping, but I still haven't got any response from you @daiyam
It would be good to know if the rounded square is just for previewing (because only Mac has guidelines) or if you want to have it on all systems.

For more details you can look at my other comments: https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-4342208 & https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-4362865

**Reply by daiyam** [reply]
@evur I've replied in that thread.

**Comment by pmahend1** [comment]
I thought we are getting rid of current `codium` because of consistency and visibility of inner shape issues.
It is still present in the poll however. Is it because some people still preferring current icon? or am I missing something? So `codium` is in top contenders. I am afraid we will retain existing icon.

**Comment by Hate9** [comment]
Hey, uhh, why is the icon we had before this (the hood-looking one) not even an option? That seems, frankly, bizarre.

**Reply by GitMensch** [reply]
If you mean the green potty icon - have a look at the discussion issue #1218 (yes, that is out of the list)

**Reply by daiyam** [reply]
I didn't find the need to add an icon which was used and due to backslash, was reverted in favor of the original one. Maybe a mistake from my part.

**Reply by GitMensch** [reply]
Nope, there were too much votes against that old icon already. We definitely shouldn't go back and with the build scripts available someone could easily fork and use GitHub actions to build with only the icon changed as post-patch.

**Reply by Hate9** [reply]
Idk, I think the icon that won all the polls originally should at least be an *option* now. Maybe that's crazy of me.

**Reply by GitMensch** [reply]
That seems related to #313/#413, right? Possibly see #491 for some of the discussions on it.

**Reply by GitMensch** [reply]
That said... I'm of course fine with adding it (but it will likely seen as "not fair", because it would be added after the first 100 votes.
I'd also be fine with a question of "change the background" (at least light/dark), but the more options added the harder these polls likely get...

**Reply by AuraHerreroRuiz** [reply]
I don't think it's bizarre, but perhaps I am biased because upon looking at that icon, I don't like it. However, I think you could also say it'd be strange to put an icon that was already rejected due to backslash, in the poll. It could come across as insistent.

**Comment by Petrosdevri** [comment]
I think current one is nice, frankly, I dislike minimalism but it's up to all of you of course.

**Reply by arihunta** [reply]
I'm curious about its story. What is it supposed to be, some kind of coral? With watching The Last of Us these days, my mind drifts to fungi 😅 . But yes it is nice.

**Reply by Petrosdevri** [reply]
Not sure bro, I guess it resembles a coral yet I like it because it's kinda innovative and special, other logos are either electrons or letters.

**Reply by bend-n** [reply]
Its a codium fragile.

[![](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse3.mm.bing.net%2Fth%3Fid%3DOIP.NlgNrKlZQ7z9vU9uEyu0BgHaE8%26pid%3DApi&f=1&ipt=77fcf3ca2f7a5bbbeff5c3526c96c140050874682c444df1be946d184654e01c&ipo=images)](https://en.wikipedia.org/wiki/Codium_fragile)

**Comment by mndscp** [comment]
I do like the derived version the best now (left-most in my screenshot below).
To help me decide I edited some icons into my application menu by simply selecting the biggest white circle in the icons from the voting image. They are all too close to the icon border in my opinion. Are you going to adjust the padding for the final icon as well? It needs to work with round icons.

![image](https://user-images.githubusercontent.com/81937645/222977106-756e28e4-35af-4910-b7d5-c26846b743b0.png)

Current icon for comparison. This also has enough padding:

![image](https://user-images.githubusercontent.com/81937645/222977231-de3602a4-3058-475e-8664-5426669e658c.png)

**Reply by daiyam** [reply]
@mndscp There won't be any white background for linux and windows (based on there guidelines, https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5200807)

**Reply by mndscp** [reply]
> @mndscp There won't be any white background for linux and windows (based on there guidelines, [#1218 (reply in thread)](https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5200807))

Thank you, I had not seen that.
Most other icons I see have some form of contrast, bright and dark colors. The simplified icon with just a gradient of similar colors looks a bit lacking here in my opinion. But maybe this is not what it's going to look like in the end. Looking forward to the re-vamp, thank you!

`</late-opinion>`

![image](https://user-images.githubusercontent.com/81937645/222979425-ab5a4ed7-5763-4d2d-bd1d-40f524637400.png)

**Reply by daiyam** [reply]
I agree. That's why I will add a border with lighter color on those platforms.
Mid-month, I'm thinking to post the possible final icons with the then most-voted shape and color, so we can try them out and tweak them if needed.

**Reply by ghost** [reply]
Reminds me of JetBrains

**Reply by geekley** [reply]
> @mndscp There won't be any white background for linux and windows (based on there guidelines, [#1218 (reply in thread)](https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5200807))

@daiyam Noooo it's awful:
![image](https://user-images.githubusercontent.com/38332912/229219139-0fec15d8-21bb-482b-b472-e5a77c77dffa.png)
It NEEDS to have a second color for contrast. Why would you remove the white background?
And it looks worse in 16px too. Please add the white square/circle whatever. And please, NOT A BORDER, it should be shaped like an app, preferably square with round corner (though circle, square and squircle is fine too).

**Reply by GitMensch** [reply]
I already start to like this on Windows. .... and if its in the system guidelines then this possibly should stay in general.
Note: @daiyam said:

> That's why I will add a border with lighter color on those platforms.

so I think there is a change upcoming that will solve the issue for you.

**Reply by geekley** [reply]
> I already start to like this on Windows. .... and if its in the system guidelines then this possibly should stay in general. Note: @daiyam said:
> 
> > That's why I will add a border with lighter color on those platforms.
> 
> so I think there is a change upcoming that will solve the issue for you.

Cool, but I'm afraid of what he means by "border". It it's the icon approved here (white rounded square) it's fine.
![image](https://user-images.githubusercontent.com/38332912/229223622-d9947244-2501-42c8-ba1d-e08f36b6d56a.png)
But if he means a white outline around the coral, that will be terrible; it should be shaped like an app.
The guidelines for KDE do recommend a "base" (background): https://develop.kde.org/hig/style/icons/colorful/#anatomy
> Icons can have a background to serve as a base for their foreground.

**Reply by daiyam** [reply]
> That's why I will add a border with lighter color on those platforms.

I forgot...

**Reply by daiyam** [reply]
After testing, borders won't do it (too small). On Windows, the app icon will use a lighter blue. I need to test on Linux...

**Comment by ghost** [comment]
Is there a specific reason the icon is getting changed?
I find the current one to be perfect.

**Reply by GitMensch** [reply]
It is problematic in some cases, see https://github.com/VSCodium/vscodium/discussions/1218 and the issues linked there.

**Comment by GitMensch** [comment]
Question: if the poll goes on as it currently does and we do get Simplified blue as an _icon_ - would there be any reason to not stay with Codium as _logo_ (used in the docs/website/as background for empty/new windows/windows installer) while switching the icon?

**Comment by MolassesLover** [comment]
I prefer the 'codium' option, though, I'm completely fine with the 'simplified' option- as long as the lines aren't **that** thick.

**Comment by tharward-r7** [comment]
I hope potential conflicts and infringement on the big-co's brands are a consideration in this. That alone makes me not want to touch derive or style icons with a 10 meter pole.

**Reply by pmahend1** [reply]
100% agree on this. Derived may touch copyright stuff if there is any.

**Comment by rafmst** [comment]
Suggestion, after the voting is over, change the default icon but still allow users to change to their taste. I'm really digging c-red but seems that is a lost haha still, would be cool to allow us to change :D

**Reply by GitMensch** [reply]
As noted in the other pol/issue: if you can provide a patch how to make this adjustable to the user, we integrate it - people have checked but found no solution, because upstream vscode does not have this as "user customization option".
You're welcome to create an issue there (possibly for switching between the blue "standard" and the green "insider" icon). If a feature like this is in, then we can provide the choice, too.

**Reply by rafmst** [reply]
I understand, my bad ;) If I had the knowledge I would do it

**Comment by daiyam** [comment]
Preview is available, see comment at https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5357728

**Comment by cavcrosby** [comment]
I like 'simplified', and actually prefer simplified's current line thickness. The current codium icon I feel is _smaller_ respectively to other app icons in my dock.

**Comment by Yanni8** [comment]
They look all amazing

**Comment by daiyam** [comment]
I will close the votes 12 hours earlier since the next version is already out (a week early). So you have until 12:00 PM UTC to cast your votes.

**Comment by daiyam** [comment]
Thank you to every voters to have participated in this vote. :clap:

@psergiojr Thank you for your winning icon  :tada: :tada:

**Reply by mateeni-dev** [reply]
Alas, I'm one of the few who voted for "style" and even fewer "yellow" option. 

Is there a permitted-use License for these branding images somewhere? Who would I give credit, if I use them in a public fork?

**Reply by daiyam** [reply]
I will try

**Comment by tharward-r7** [comment]
So this is the one then?

<img src="https://user-images.githubusercontent.com/57955862/229143220-d9c01168-1e38-4093-88d2-3a4ac251bb2b.png" width="100">

**Comment by geekley** [comment]
Ah too bad I never knew of this poll before the update (wasn't there a way to notify users of the poll in-app?).
I think the ideal would have been the old icon with the new simplified icon only on small sizes, i.e. 24px and below.
![image](https://user-images.githubusercontent.com/38332912/229224847-e0416630-9a3f-4fe8-86c5-8724b12e03c7.png) ![image](https://user-images.githubusercontent.com/38332912/229224672-bc852584-7718-4cc2-a94e-4df3698ea1e6.png) ![image](https://user-images.githubusercontent.com/38332912/229223622-d9947244-2501-42c8-ba1d-e08f36b6d56a.png)

**Reply by GitMensch** [reply]
> wasn't there a way to notify users of the poll in-app?

Totally, and it stand there on every of your empty vscodium windows on the welcome page on the lower left...

**Reply by geekley** [reply]
Ah, only now I understand what you mean. There's an "announcements section", apparently? Though I've never noticed it before. It appears only when recent is empty? Sorry, are you saying this *was* done or that it *could have been* done?
In any case, I was thinking of something more prominent like a notification or a tab similar to "release notes" opening on its own.

**Reply by GitMensch** [reply]
nope - directly on the welcome page - erything else would have been to intrusive - see https://github.com/VSCodium/vscodium/pull/1274 for that updated (shipped with October/November) and a screenshot that has "new icon" in the announcement. A later version did add icons before and included both polls.

**Comment by geekley** [comment]
For the ones who got too used to the circle BG and don't wanna change muscle memory too much:
![codium-stable-circle](https://user-images.githubusercontent.com/38332912/231310041-b299793d-6e04-4276-b05e-fc3296a19253.png) ![](https://user-images.githubusercontent.com/38332912/231311170-61dc67f3-07ea-411c-9dfb-e25928f45094.png) ![](https://user-images.githubusercontent.com/38332912/231311175-4faa1cff-1078-43e1-b6ed-3059f2c4c110.png) ![](https://user-images.githubusercontent.com/38332912/231311182-59fe11d0-03f7-4bd6-98ae-e503428e8841.png)
SVG here: https://github.com/VSCodium/vscodium/discussions/1218#discussioncomment-5586517
